e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+        )
+        # `loc_container` does not need to contain checked items
+        # But we should have `self.loc` be `self.loc_selected`
+        self.loc = self.loc_container.locator("label input[type=radio]:checked")
+        # Same value
+        self.loc_selected = self.loc
+
+        self.loc_choices = self.loc_container.locator("label input[type=radio]")
+        self.loc_choice_labels = self.loc_container.locator(
+            "label",
+            has=self.page.locator("input[type=radio]"),
+        )
+
+    def set(
+        self,
+        selected: str,
+        *,
+        timeout: Timeout = None,
+        **kwargs: typing.Any,
+    ) -> None:
+
+        self.loc_container.locator(
+            f"label input[type=radio][value='{selected}']"
+        ).check(timeout=timeout)
+
+    def expect_choices(
+        self,
+        # TODO; support patterns?
+        choices: typing.List[str],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        self._expect_locator_values_in_list(
+            "input[type=radio]",
+            "choices",
+            choices,
+            timeout=timeout,
+        )
+
+    def expect_selected(
+        self,
+        selected: TextValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        # Playwright doesn't like lists of size 0. Instead, use `None`
+        if selected is None:
+            playwright_expect(self.loc_selected).to_have_count(0, timeout=timeout)
+            return
+
+        playwright_expect(self.loc_selected).to_have_value(selected, timeout=timeout)
+
+
+class InputFile(
+    # _Placeholder,
+    _InputWithLabel,
+):
+    # id: str,
+    # label: TagChildArg,
+    # *,
+    # multiple: bool = False,
+    # accept: Optional[Union[str, List[str]]] = None,
+    # width: Optional[str] = None,
+    # button_label: str = "Browse...",
+    # placeholder: str = "No file selected",
+    # capture: Optional[Literal["environment", "user"]] = None,
+    # with page.expect_file_chooser() as fc_info:
+    #     page.get_by_text("Upload").click()
+    # file_chooser = fc_info.value
+    # file_chooser.set_files("myfile.pdf")
+    def __init__(
+        self,
+        page: Page,
+        id: str,
+    ) -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"input[type=file]#{id}",
+            loc_label=f"label[id={id}-label]",
+        )
+        self.loc_file_input = self.loc
+        self.loc_button = self.loc_container.locator("label span.btn")
+        self.loc_file_display = self.loc_container.locator("input[type=text]")
+
+    def set(
+        self,
+        file_path: typing.Union[
+            str,
+            pathlib.Path,
+            FilePayload,
+            typing.List[typing.Union[str, pathlib.Path]],
+            typing.List[FilePayload],
+        ],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        self.loc.set_input_files(file_path, timeout=timeout)
```

Comment:
By default, call `self.expect_complete`. This will probably want 1) a different timeout than `timeout`, and 2) to have a way to opt out.